cloudify_cli/commands/agents.py

```diff
@@ -133,9 +135,13 @@ def worker(dep_id):
                for dep_id in deps]
 
     for t in threads:
+        t.daemon = True
         t.start()
-    for t in threads:
-        t.join()
+
+    while True:
```

Comment:
No timeout handling is needed here, because all the worker threads already have a timeout - so the whole operation will time out when all the worker thread executions do.